pmlc/util/enums.h

```diff
@@ -58,4 +59,27 @@ inline DataType promoteTypes(DataType lhs, DataType rhs) {  //
   return std::max(lhs, rhs);
 }
 
+inline DataType from_string(const std::string& dtype) {
+  static std::unordered_map<std::string, DataType> long_str_to_type{
+      {"bool", DataType::u1},       {"int8", DataType::i8},          {"uint8", DataType::u8},
+      {"int16", DataType::i16},     {"uint16", DataType::u16},       {"int32", DataType::i32},
+      {"uint32", DataType::u32},    {"int64", DataType::i64},        {"uint64", DataType::u64},
+      {"bfloat16", DataType::bf16}, {"float16", DataType::f16},      {"float32", DataType::f32},
+      {"float64", DataType::f64},   {"<invalid>", DataType::invalid}};
+
+  static std::unordered_map<std::string, DataType> short_str_to_type{
```

Comment:
I don't think we need to have a 'long' vs 'short' string name for the internal DataType. I do think it's possible that the `plaidml::DType` might differ from the `pmlc::util::DataType` (since they live in different modules, at different layers of the system, one is user facing (the interface) and the other is part of the implementation that users would never see).

Reply:
Currently, `to_string(plaidml::DType)` is not parseable by `pmlc::util::symbolizeDataType` (e.g. `float16` vs `float16`). Do you mean we should change `to_string(plaidml::DType)` to match that the format of `DataType`? A common string representation between `DataType` and `DType` seems to be the cleanest way of keeping the `plaidml` and `pmlc` APIs separate

Reply:
I'm curious why we even need to use string values anywhere other than for printing purposes. I don't think we should have users select the dtype based on a string, they should use the enum instead.